src/common

```diff
@@ -227,12 +227,39 @@ OBJ += $(patsubst %.cc, %$(OBJEXT), $(filter %.cc, $(SRC)))
 %.obj:%.c
 	$(CC) $(CP_CFLAGS) /nologo /c /EHsc $< /Fo$@
 
+# get version from git
+GIT_INFO = $(shell git describe --tags --always --dirty || echo "n/a")
+
+CBMC_VERSION_FILES += $(filter %_parse_options$(OBJEXT), $(OBJ))
+
+ifeq ($(BUILD_ENV_),MSVC)
+$(CBMC_VERSION_FILES): CP_CXXFLAGS += /DCBMC_VERSION='"""$(CBMC_VERSION) ($(GIT_INFO))"""'
+else
+$(CBMC_VERSION_FILES): CP_CXXFLAGS += -DCBMC_VERSION="\"$(CBMC_VERSION) ($(GIT_INFO))\""
+endif
+
+# Use make >= 4.0's `file` function, if available - it should be faster than
+# `shell cat`
+ifeq ($(firstword $(subst ., , $(MAKE_VERSION))), 3)
+KNOWN_RELEASE_INFO = $(shell cat .release_info 2>/dev/null)
+else
+KNOWN_RELEASE_INFO = $(file < .release_info)
+endif
+ifneq ($(GIT_INFO), $(KNOWN_RELEASE_INFO))
+$(CBMC_VERSION_FILES): .release_info
+
+.release_info:
+	echo $(GIT_INFO) > $@
+
+.PHONY: .release_info
+endif
+
 clean:
 	$(RM) $(patsubst %.cpp, %$(OBJEXT), $(filter %.cpp, $(SRC))) \
 		$(patsubst %.cpp, %$(DEPEXT), $(filter %.cpp, $(SRC))) \
 		$(patsubst %.cc, %$(OBJEXT), $(filter %.cc, $(SRC))) \
 		$(patsubst %.cc, %$(DEPEXT), $(filter %.cc, $(SRC))) \
-		$(CLEANFILES)
+		$(CLEANFILES) .release_info
```

Comment:
Very minor nits - would it be any cleaner (no pun intended...) if you just do a `CLEANFILES += .release_info` somewhere around line 233 instead? Also, is there any milage in using a variable to hold the name of `.release_info`, as that file name is used in a number of different places in the previous patch hunk? Personally I'm not sure, but thought I'd float the idea anyway.

Reply:
I have introduced a variable instead of the literal file name, but I've refrained from adding it to `CLEANFILES` as that would have seemed to be just additional code (with the risk of fiddling with a variable that other Makefiles may set).